RELEASES.md

```diff
@@ -1,3 +1,123 @@
+- [Function pointers can now be created, cast, and passed around in a `const fn`][93827]
+- [Recursive calls can now set the value of a function's opaque `impl Trait` return type][94081]
+
+Compiler
+--------
+
+- [Linking modifier syntax in `#[link]` attributes and on the command line, as well as the `whole-archive` modifier specifically, are now supported][93901]
+- [Update to LLVM 14.0.0][95247]
+- [The `char` type is now described as UTF-8 (`DW_ATE_UTF`) in DWARF debuginfo][89887]
+- The [`#[target_feature]`][target_feature] attribute [can now be used with aarch64 features][90621]
+- [`#[target_feature = "adx"]` is now stable][93745]
+- [Catching a second unwind from FFI code while cleaning up from a Rust panic now causes the process to abort][92911]
+
+Libraries
+---------
+
+- [`ManuallyDrop<T>` is now documented to have the same layout as `T`][88375]
+- [Windows paths longer than 260 chars are now supported in `process::Command` without needing explicit canonoicalization][92519]
+- [`#[ignore = "…"]` messages are printed when running tests][92714]
+- [Consistently present absent stdio handles on Windows as NULL handles][93263]
+- [Make `std::io::stdio::lock()` return `'static` handles][93965]
+- [`Vec::from_raw_parts` is now less restrictive about its inputs][95016]
+- [Use cgroup quotas for calculating `available_parallelism` on Linux][92697]
+
+Stabilized APIs
+---------------
+
+- [`From<&[T; N]>` for `Vec<T>`][95098]
+- [`From<&mut [T; N]>` for `Vec<T>`][95098]
+- [`Pin::static_mut`][93580]
+- [`Pin::static_ref`][93580]
+- [`Vec::retain_mut`][95491]
+- [`VecDeque::retain_mut`][95491]
+- [`Write` for `Cursor<[u8; N]>`][92663]
+- [`std::os::unix::net::SocketAddr::from_pathname`][94356]
+- [`std::process::ExitCode`][93840]
+- [`std::process::Termination`][93840]
+- [`std::thread::JoinHandle::is_finished`][95130]
+
+These APIs are now usable in const contexts:
+
+- [`*const T::offset` and `*mut T::offset`][93957]
+- [`*const T::wrapping_offset` and `*mut T::wrapping_offset`][93957]
+- [`*const T::add` and `*mut T::add`][93957]
+- [`*const T::sub` and `*mut T::sub`][93957]
+- [`*const T::wrapping_add` and `*mut T::wrapping_add`][93957]
+- [`*const T::wrapping_sub` and `*mut T::wrapping_sub`][93957]
+- [`[T]::as_mut_ptr`][93957]
+- [`[T]::as_ptr_range`][93957]
+- [`[T]::as_mut_ptr_range`][93957]
+
+Cargo
+-----
+
+No feature changes, but see compatibility notes.
+
+Compatibility Notes
+-------------------
+
+- [Disable `#[thread_local]` support on i686-pc-windows-msvc][95430]
+- [Proc macros no longer see `ident` matchers wrapped in groups][92472]
+- [The number of `#` in `r#` raw string literals is now required to be less than 256][95251]
+- [When checking that a dyn type satisfies a trait bound, supertrait bounds are now enforced][92285]
+- [`cargo vendor` now only accepts one value for each `--sync` flag][cargo/10448]
+- [`cfg` predicates in `all()` and `any()` are always evaluated to detect errors, instead of short-circuiting][94295]
```

Comment:
```suggestion
- [`cfg` predicates in `all()` and `any()` are always evaluated to detect errors, instead of short-circuiting.][94295] The compatibility considerations here arise in nightly-only code that used the short-circuiting behavior of `all` to write something like `cfg(all(feature = "nightly", syntax-requiring-nightly))`, which will now fail to compile. Instead, use either `cfg_attr(feature = "nightly", ...)` or nested uses of `cfg`.
```